File: Problem-052/solution.js
function permutedMultiples(n) {
  function sameDigit(a,b){
    return a.toString().split('').sort().join('')=== b.toString().split('').sort().join('');
  }
   let x = 1;
   while(true){
    let found = true;
    for (let multiplier = 2; multiplier <= n; multiplier ++){
      if ( !sameDigit(x, x*multiplier)){
        found = false;
        break;
      }
    }
    if (found) return x;
    x++;
   }
}

File: Problem-052/solution.py
def permuted_multiples(n):

    def same_digit(a, b):
        return sorted(str(a)) == sorted(str(b))

    x = 1

    while True:

        found = True

        for multiplier in range(2, n + 1):

            if not same_digit(x, x * multiplier):
                found = False
                break

        if found:
            return x

        
        x += 1


